#! /bin/sh -e
## 20_qmail_to_mailman.debian.dpatch by Tollef Fog Heen <tfheen@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix some paths in the qmail_to_mailman wrapper

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch ${2:+-d $2}}"

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $patch_opts -p1 < $0;;
    -unpatch) patch $patch_opts -p1 -R < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0
@DPATCH@

--- mailman-2.1.4.orig/contrib/qmail-to-mailman.py
+++ mailman-2.1.4/contrib/qmail-to-mailman.py
@@ -28,12 +28,12 @@
 #
 # INSTALLATION:
 #
-# Install this file as ~mailman/qmail-to-mailman.py
+# Install this file as ~mailman/bin/qmail-to-mailman.py
 #
 # To configure a virtual domain to connect to mailman, create these files:
 #
 # ~mailman/.qmail-default
-# |preline @PYTHON@ @prefix@/mail-in.py
+# |preline /usr/bin/python /usr/lib/mailman/bin/mail-in.py
 #
 # /var/qmail/control/virtualdomains:
 # DOMAIN.COM:mailman
@@ -64,7 +64,7 @@
         sys.exit(100)
 
     local = string.lower(local)
-    local = re.sub("^mailman-","",local)
+    local = re.sub("^" + MailmanUser + "-","",local)
 
     names = ("root", "postmaster", "mailer-daemon", "mailman-owner", "owner",
              "abuse")
